File: rdf4j/src/main/scala/eu/ostrzyciel/jelly/convert/rdf4j/Rdf4jConverterFactory.scala
package eu.ostrzyciel.jelly.convert.rdf4j

import eu.ostrzyciel.jelly.core.{ConverterFactory, ProtoEncoder}
import org.eclipse.rdf4j.model.{Statement, Value}

object Rdf4jConverterFactory
  extends ConverterFactory[Rdf4jEncoderConverter, Rdf4jDecoderConverter, Value, Rdf4jDatatype, Statement, Statement]:

  override lazy val encoderConverter: Rdf4jEncoderConverter = new Rdf4jEncoderConverter()

  /**
   * @inheritdoc
   */
  override lazy val decoderConverter: Rdf4jDecoderConverter = new Rdf4jDecoderConverter()

File: rdf4j-java/src/main/java/eu/neverblink/jelly/convert/rdf4j/Rdf4jConverterFactory.java
package eu.neverblink.jelly.convert.rdf4j;

import eu.neverblink.jelly.core.JellyConverterFactory;
import org.eclipse.rdf4j.model.Value;

/**
 * A singleton factory for creating Jelly RDF4J converters.
 * <p>
 * This class is a singleton and should be accessed via the {@link #getInstance()} method.
 */
public final class Rdf4jConverterFactory
    extends JellyConverterFactory<Value, Rdf4jDatatype, Rdf4jEncoderConverter, Rdf4jDecoderConverter> {

    private static final Rdf4jConverterFactory INSTANCE = new Rdf4jConverterFactory();

    private Rdf4jConverterFactory() {}

    /**
     * Returns the singleton instance of the {@link Rdf4jConverterFactory}.
     *
     * @return the singleton instance of the {@link Rdf4jConverterFactory}
     */
    public static Rdf4jConverterFactory getInstance() {
        return INSTANCE;
    }

    @Override
    public Rdf4jEncoderConverter encoderConverter() {
        return new Rdf4jEncoderConverter();
    }

    @Override
    public Rdf4jDecoderConverter decoderConverter() {
        return new Rdf4jDecoderConverter();
    }
}
